.github/workflows/darwin.yaml

```diff
@@ -122,32 +106,14 @@ jobs:
               timeout-minutes: 15
               run: |
                   scripts/examples/gn_build_example.sh examples/all-clusters-app/linux out/debug chip_config_network_layer_ble=false
-              env:
```

Comment:
So this is actually losing us test coverage.

Right now this workflow is testing that the xcodebuild steps succeed without ZAP available.  That's a must-have, and we added CI for it specifically because it got broken at one point, but that was covered up by blanket-installing ZAP for this workflow.  We need to make sure it does not ever get broken again.  That's why right now only some of the workflow steps here have ZAP available.

The xcodebuild steps here _must_ function in an environment with no cipd anything, in particular (and no bootstrap/activate, for that matter).

Any thoughts on how we can keep testing what we're testing right now?

Reply:
Maybe for the xcodebuild steps we could set `ZAP_INSTALL_PATH` or `ZAP_DEVELOPMENT_PATH` to somewhere that does not in fact have ZAP or something....  I'd still be worried about `zap_execution.py` deciding to be "smart" and falling back to `$PATH` if it doesn't find ZAP where it's being pointed to.

Reply:
We could find zap-cli and rename it. Would that work for you? like `which zap-cli` moved to zap-cli.moved and then move back once done.

Which step should I guard like that?

Reply:
It think that would work, yes.  All the steps that do xcodebuild should be guarded like that.

Reply:
And ideally in each step where we do that we then try to invoke `zap-cli` and check that this fails.

Reply:
Added move (and undo) and a separate check that zap-cli does indeed fail.